src/pages/Compare/index.tsx

```diff
@@ -66,45 +73,55 @@ export class CompareBase extends React.Component<Props> {
       return;
     }
 
-    this.loadData();
-  }
+    const path = getPathFromQueryString(history);
 
-  componentDidUpdate(prevProps: Props) {
-    this.loadData(prevProps);
-  }
-
-  loadData(prevProps?: Props) {
-    const { match, path } = this.props;
-    const { addonId, baseVersionId, headVersionId } = match.params;
+    if (!version) {
+      dispatch(
+        _fetchDiff({
+          addonId: parseInt(addonId, 10),
+          baseVersionId: oldVersionId,
+          headVersionId: newVersionId,
+          path: path || undefined,
+        }),
+      );
+      return;
+    }
 
-    if (
-      !prevProps ||
-      (prevProps.path && path !== prevProps.path) ||
+    if (!prevProps) {
+      if (path && path !== version.selectedPath) {
+        // We preserve the hash in the URL (if any) when we load the file from
+        // an URL that has likely be shared.
+        this.viewVersionFile(path, { preserveHash: true });
+      }
+    } else if (
+      (prevProps.version &&
+        version.selectedPath !== prevProps.version.selectedPath) ||
```

Comment:
I guess nothing here is broken but we should update this someday to not use `prevProps` values for comparison. It is fragile. The best way to do it is to compare Redux mapped properties.